architecture/cluster-fulfillment.md

```diff
@@ -1 +1,319 @@
+   - **ServiceAccount**: Identity for cluster-related automation
+   - **RoleBindings**: RBAC permissions for the service account to manage cluster resources
+
+3. **EDA Webhook Trigger**: Calls the Event Driven Ansible webhook endpoint to initiate cluster provisioning:
+   - The webhook payload includes the complete ClusterOrder specification.
+
+4. **HostedCluster Monitoring**: After the webhook is triggered, the controller watches for a HostedCluster resource (created by the Ansible automation) and monitors its conditions:
+   - Waits for the control plane to become available
+   - Monitors that the cluster is not degraded
+   - Tracks node pool readiness
+
+5. **Status Synchronization**: Continuously updates the ClusterOrder status based on the HostedCluster state, including:
+   - Phase transitions (Progressing → Ready or Failed)
+   - Condition updates
+   - Cluster reference information (namespace, resource names)
+
+6. **Deletion Handling**: When a ClusterOrder is deleted:
+   - Triggers the deletion webhook to EDA
+   - Ensures all associated cluster resources are cleaned up
+   - Uses finalizers to prevent premature deletion
+
+### Ansible Automation Platform - Cluster Provisioning
+
+Event Driven Ansible (EDA) running on AAP receives webhook events from the OSAC
+Controller and orchestrates the actual cluster provisioning.
+
+**EDA Rulebook** (`cloudkit-aap/rulebooks/cluster_fulfillment.yml`):
+
+The rulebook listens on port 5000 and defines rules for cluster lifecycle events:
+- Endpoint: `create-hosted-cluster` → Triggers the cluster creation workflow
+- Endpoint: `delete-hosted-cluster` → Triggers the cluster deletion job
+
+When triggered, EDA launches the appropriate workflow template in AAP.
+
+**Cluster Creation Playbook** (`cloudkit-aap/playbook_cloudkit_create_hosted_cluster.yml`):
+
+The main cluster creation workflow consists of these phases:
+
+1. **Retrieve Cluster Order Details**:
+   - Fetches the ClusterOrder object from Kubernetes
+   - Generates a unique lock holder ID to prevent concurrent modifications
+   - Applies default cluster settings via the `cloudkit.service.cluster_settings` role
+
+2. **Infrastructure Preparation**:
+   - Extracts template information from the ClusterOrder
+   - Determines the working namespace
+   - Acquires a cluster lock to ensure safe concurrent operations
+   - Adds an infrastructure finalizer to the ClusterOrder to track cleanup
+
+3. **Template Execution**:
+   - Dynamically includes the selected cluster template's `install` tasks
+   - Templates are Ansible roles located in `cloudkit-aap/collections/ansible_collections/cloudkit/templates/roles/`
+   - Each template provides a standardized interface while allowing customization of the underlying implementation
+
+4. **Cluster Infrastructure Creation**:
+   - The `cloudkit.service.cluster_infra` role manages the hosted cluster lifecycle
+   - Creates or updates a HostedCluster resource
```

Comment:
The `cluster_infra` role does not create or update a HostedCluster resource (that's done by the `hosted_cluster` role). The `cluster_infra` task is responsible for configuring networking and allocating nodes.